sklearn/utils/validation.py

```diff
@@ -494,7 +498,15 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         with warnings.catch_warnings():
             try:
                 warnings.simplefilter('error', ComplexWarning)
-                array = np.asarray(array, dtype=dtype, order=order)
+                array = np.asarray(array, order=order)
```

Comment:
I've not checked the full context, but can this do the wrong thing if dtype=object, and some data elements are strings?

Reply:
Indeed. Reverted to the previous behavior by default which should fix test failures.